test/test_integration.py

```diff
@@ -44,6 +45,10 @@ def test_spin_backend() -> None:
         """,
     )
 
+    # Check whether the above algorithm can be mapped to a dummy chip topology
+    connectivity = {0: [1, 2], 1: [0], 2: [0, 3], 3: [2]}
+
+    qc.route(interactions=[(0, 1), (2, 3), (0, 2)], router=RoutingChecker(connectivity))
```

Comment:
If the `route` methods in the various places (`Circuit.route`, `Router.route`, and `RoutingChecker.route`) is adjusted to accepting the IR as input, rather than the interactions, this could be simplified to:
 ```suggestion
    qc.route(router=RoutingChecker(connectivity))
```